### PR TITLE
fix: standardizing the quick-start of the Kotlin school timetabling module

### PR DESCRIPTION
**Pull request by zepfred**
The pull request updates the quick-start of the Kotlin school timetabling module. All classes from the Quarkus project were adapted and converted to the Kotlin language.

solver1
solver2
solver3

Issue #182

### CODE REVIEW

File: technology/kotlin-quarkus/src/main/kotlin/org/acme/kotlin/schooltimetabling/domain/Timetable.kt [hunk 1]
```diff
@@ -9,16 +9,18 @@ import ai.timefold.solver.core.api.score.buildin.hardsoft.HardSoftScore
 import ai.timefold.solver.core.api.solver.SolverStatus
 
 @PlanningSolution
-class TimeTable {
+class Timetable {
```
Comment (triceo) [inline technology/kotlin-quarkus/src/main/kotlin/org/acme/kotlin/schooltimetabling/domain/Timetable.kt:12]: Nice detail, well spotted.

File: technology/kotlin-quarkus/src/main/kotlin/org/acme/kotlin/schooltimetabling/rest/TimetableDemoResource.kt [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+    fun generate(
+        @Parameter(
+            description = "Unique identifier of the demo data.", required = true
+        ) @PathParam("demoDataId") demoData: DemoData
+    ): Response {
+        val timeslotList: MutableList<Timeslot> = ArrayList(10)
+        var nextTimeslotId = 0L
+        timeslotList.add(
+            Timeslot(nextTimeslotId++, DayOfWeek.MONDAY, LocalTime.of(8, 30), LocalTime.of(9, 30))
+        )
```
Comment (triceo) [inline technology/kotlin-quarkus/src/main/kotlin/org/acme/kotlin/schooltimetabling/rest/TimetableDemoResource.kt]: This is very verbose. Some later calls take even 5 lines.
Maybe by static importing `DayOfWeek` values, we can get this to fit on one line?

File: technology/kotlin-quarkus/src/main/kotlin/org/acme/kotlin/schooltimetabling/rest/TimetableResource.kt [hunk 3]
```diff
@@ -0,0 +1,255 @@
+package org.acme.kotlin.schooltimetabling.rest
+
+import ai.timefold.solver.core.api.score.analysis.ScoreAnalysis
+import ai.timefold.solver.core.api.score.buildin.hardsoft.HardSoftScore
+import ai.timefold.solver.core.api.solver.ScoreAnalysisFetchPolicy
+import ai.timefold.solver.core.api.solver.SolutionManager
+import ai.timefold.solver.core.api.solver.SolverManager
+import jakarta.inject.Inject
+import jakarta.ws.rs.*
```
Comment (triceo) [inline technology/kotlin-quarkus/src/main/kotlin/org/acme/kotlin/schooltimetabling/rest/TimetableResource.kt]: Please configure your IDE to never use `*` imports.
(One of our little rules; makes reviews easier, and creates less classloading work for the JVM.)

File: technology/kotlin-quarkus/src/main/kotlin/org/acme/kotlin/schooltimetabling/solver/TimeTableConstraintProvider.kt [hunk 4]
```diff
@@ -48,6 +56,13 @@ class TimeTableConstraintProvider : ConstraintProvider {
                 Joiners.equal(Lesson::teacher)
             )
             .penalize(HardSoftScore.ONE_HARD)
+            .justifyWith({ lesson1: Lesson, lesson2: Lesson?, score: HardSoftScore? ->
+                TeacherConflictJustification(
+                    lesson1.teacher,
+                    lesson1,
+                    lesson2!!
+                )
```
Comment (triceo) [inline technology/kotlin-quarkus/src/main/kotlin/org/acme/kotlin/schooltimetabling/solver/TimeTableConstraintProvider.kt]: In the interest of brevity, I'd put this on one line.

**Review by triceo — APPROVED** [review]
Nice! I'm not a Kotlin expert, I'm learning as I go, but LGTM.

I'm making some cosmetic suggestions.
Also, did I miss the update of the README?

**Comment by zepfred** [comment]
@triceo, I have tagged you once again to confirm the changes made in the file `README.adoc`. Since we are no longer storing state, I have included a parameter to remove the container after stopping it (`--rm`) and changed the default image path to `$USER/timefold-solver-kotlin-quarkus-school-timetabling-quickstart:1.0-SNAPSHOT`.

**Review by triceo — APPROVED** [review]
Works for me!

**Pull request merged**